addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -1252,8 +1252,8 @@ export const formatSelection = (editor, formatName, {applyStyle, formatProps} =
         }
     }
 }
-export const isLinkEligibleForZwnbsp = link => {
-    return link.isContentEditable && !(
+export const isLinkEligibleForZwnbsp = (editable, link) => {
+    return link.isContentEditable && editable.contains(link) && !(
```

Comment:
But it shouldn't happen anyway that we give a link that isn't in the editable. When would we be doing that without this check?

Reply:
In `resetLinkInSelection`, if we don't check that the selection is in the editable. It seemed more safe to me to check here than there, in case we add more calls to this later. It also makes sense to me since this is giving the conditions for applying the link padding, and one of these conditions is that the link is in the editable area.

Reply:
I'm ok to keep the guard but IMO we should fix `resetLinkInSelection` too to check that the selection is in the editable, because otherwise I feel like the function is just wrong.

Reply:
I just checked and actually we don't check in `insert` either if we're inserting in the editable or not. Should I add that check at the very top of `insert` too then?